### PR TITLE
Add MANUAL_CONTROL handling with new plugin

### PR DESCRIPTION
**Pull request by v01d**
Cleaned up pull request #377 (I messed up my repo in the process of fixing).

### CODE REVIEW

File: mavros/src/plugins/manual_control.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	ros::Publisher control_pub;
+
+	/* -*- rx handlers -*- */
+
+	void handle_manual_control(const mavlink_message_t *msg, uint8_t sysid, uint8_t compid) {
+		mavlink_manual_control_t manual_control;
+		mavlink_msg_manual_control_decode(msg, &manual_control);
+
+		mavros_msgs::ManualControl manual_control_ros;
+		manual_control_ros.header.stamp = ros::Time::now();	// correct?
```
Comment (vooon) [inline mavros/src/plugins/manual_control.cpp]: Yes, correct because message don't have any timestamp.

**Comment by vooon** [comment]
Ok, now it looks good to merge. But maybe you implement send code too?

**Comment by v01d** [comment]
I'm not aware that this mavlink message goes both ways. I think it is only
sent by the autopilot, and not received. But I'm not sure.

**Comment by vooon** [comment]
You may check QGC code. In SITL/HIL mode it has ability to send joystick data. I think via that message.

**Comment by v01d** [comment]
To be honest, at the moment I'm not sure If I can manage time to look into
that. I can do it bit further on and provide a separate pull request.

**Comment by vooon** [comment]
Ok.

**Comment by v01d** [comment]
@vooon I'm just testing this using running mavros on my odroid connected to the pixhawk via the TELEM port and I can't see the MANUAL_CONTROL stream there. However, if I connect to the pixhawk from my laptop using USB cable, I do see the stream. Do you know what could be the cause?

**Comment by vooon** [comment]
I think stream rate not configured on TELEM. See [`rc.usb`][1] file.

[1]: https://github.com/PX4/Firmware/blob/9365c5a4383d8702e66b5f0e3f95a1acb3fd5cac/ROMFS/px4fmu_common/init.d/rc.usb

**Comment by vooon** [comment]
Hmm, now stream config for usb is moved to mavlink app, but you still can configure streams via `extras.txt`

**Comment by v01d** [comment]
Ok, thanks!

On Mon, Aug 31, 2015 at 12:24 PM, Vladimir Ermakov <notifications@github.com

**Pull request merged**